flocker/ca/_ca.py

```diff
@@ -0,0 +1,53 @@
+# Copyright Hybrid Logic Ltd.  See LICENSE file for details.
+
+"""
+Low-level logic for a certificate authority.
+
+Uses ECDSA + SHA256? Or if that turns out to be too difficult RSA 4096-bit
++ SHA 256.
+"""
+
+from twisted.internet.ssl import KeyPair
+
+
+def generate_keypair():
+    """
+    Create a new key pair.
+    """
+    #return KeyPair.generate(crypto.TYPE_ECDSA, size=something?)
+    pass
+
+
+class CertificateAuthority(object):
+    """
+    A certificate authority whose configuration is stored in a specified
+    directory.
+    """
+    def __init__(self, path):
+        """
+        :param FilePath path: Directory where private key and certificate are
+            stored.
+        """
+        # try to load private key and certificate, raise exception if that
+        # fails.
+
+    @classmethod
+    def initialize(cls, path, name):
+        """
+        Generate new private/public key pair and self-sign, then store in
+        given directory.
+
+        :param FilePath path: Directory where private key and certificate are
+            stored.
+        :param bytes name: The name of the cluster.
```

Comment:
If this is to identify the certificate subject, probably better to accept a `DistinguishedName` here (or a `dict` to turn in to one, at least).  The purpose of a single bare "name" string might be easily misunderstood.

Do we even want a `commonName`?  The CA itself may not be an addressable entity in this module (it's only used by our command line tool).  Organization name seems relevant.  Perhaps also organizational unit.  Maybe some location stuff but maybe we don't actually need to be that featureful.

Other considerations - certain extensions would be relevant for this certificate, for example the CA:true constraint (random example code, <http://stackoverflow.com/questions/14972345/creating-self-signed-certificate-using-pyopenssl>).  This is possibly worth a follow-up ticket, maybe not relevant for now though (I'm **somewhat** sure forgetting about these won't cause validation issue...)